lookup_handler.go

```diff
@@ -0,0 +1,76 @@
+package ctrl
+
+import (
+	"github.com/atlassian/ctrl/logz"
+	"go.uber.org/zap"
+	meta_v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/tools/cache"
+)
+
+// LookupHandler is a handler for objects that can be mapped to some parent object
+// through the use of an SharedIndexInformer.
+type LookupHandler struct {
+	Logger       *zap.Logger
+	WorkQueue    WorkQueueProducer
+	ZapNameField ZapNameField
+
+	Lookup func(runtime.Object) ([]runtime.Object, error)
+}
+
+// runtime.Object
+func (e *LookupHandler) enqueueMapped(logger *zap.Logger, obj meta_v1.Object) {
+	objs, err := e.Lookup(obj.(runtime.Object))
+	if err != nil {
+		logger.Info("Failed to get objects from index")
```

Comment:
`logger.Error("Failed to get objects from index", zap.Error(err))`. Log it as error because it should not happen and also don't forget to attach the actual error.